### ISSUE TITLE
TimeLock Agent: remoting2 -> remoting3, Observable -> Supplier

### ISSUE THREAD

**Issue by jeremyk-91** [palantir/atlasdb#2338]
Internal server framework requires these.

@tpetracca, @tboam for SA - I'm prioritising this quite highly (despite the issue name sounding innocuous) because it, for various reasons, prevents us from properly rendering metrics from the TimeLock Server.

**Comment by nziebart** [comment]
I looked at trying to convert `Refreshable` to `Observable` but it wasn't immediately obvious how to do so.. I wonder if we can just take a supplier

**Comment by jeremyk-91** [comment]
Yep, we should go with Supplier for now.

Regarding conversion, this gets close, but I suspect has concurrency problems:
```
    private static <T> Observable<T> fromRefreshable(Refreshable<T> refreshable) {
        return Observable.create(emitter -> {
            emitter.onNext(refreshable.get());
            refreshable.subscribe(emitter::onNext);
        });
    }
```

Eventually we might want the "proper" publish/subscribe (I can see how it could be useful in graphlock) but we can do that when we need to.

**Comment by nziebart** [comment]
@jeremyk-91 are you working on bumping to remoting3 as part of this also?

if so that could take the place of https://github.com/palantir/atlasdb/pull/2343

**Comment by jeremyk-91** [comment]
(a bit late, but yes; #2348 is the remoting3 PR!)

**Comment by gsheasby** [comment]
Fixed by #2339 and #2348.
